engine/src/main/java/org/destinationsol/game/asteroid/AsteroidBuilder.java

```diff
@@ -32,20 +33,48 @@
 import org.destinationsol.game.drawables.Drawable;
 import org.destinationsol.game.drawables.DrawableLevel;
 import org.destinationsol.game.drawables.RectSprite;
+import org.json.JSONObject;
+import org.terasology.assets.ResourceUrn;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
 
 public class AsteroidBuilder {
     private static final float DENSITY = 10f;
     private static final float MAX_A_ROT_SPD = .5f;
     private static final float MAX_BALL_SZ = .2f;
     private final CollisionMeshLoader collisionMeshLoader;
     private final List<TextureAtlas.AtlasRegion> textures;
+    private final String moduleName;
```

Comment:
Why does an AsteroidBuilder need a moduleName?

Reply:
Firstly, since the textures and collision meshes are loaded in this file, it needs a module to find the assets from. Secondly, the Asteroid constructor requires a moduleName param, hence AsteroidBuilder needs to store this variable globally to be used within the class.